ocs/agent/aggregator.py

```diff
@@ -149,10 +150,174 @@ def empty(self):
+            >>> new_name = Provider._check_for_duplicate_names(name, current_field_names)
+            >>> print(new_name)
+            test_02
+
+        Args:
+            field_name (str): field name to check against name_lsit
+            name_list (list): list of field names already in a Block
+
+        Returns:
+            str: A new field name that is not already in name_list
+
+        """
+        # Assume duplicate name, until proven otherwise
+        duplicate_name = True
+        name_index = 1
+
+        while duplicate_name:
+            if field_name not in name_list:
+                duplicate_name = False
+            else:
+                if len(field_name) < 252:
+                    field_name += f'_{name_index:02}'
+                else:
+                    field_name[:-3] += f'_{name_index:02}'
+
+            name_index += 1
+
+        return field_name
+
+    def _rebuild_invalid_data(self, data):
+        """Rebuild an invalid data dictionary.
+
+        Args:
+            data (dict): data dictionary passed to Provider.save_to_block().
+
+        Returns:
+            dict: A rebuilt data dictionary with invalid characters stripped
+                  from the field names, limited to 255 characters in length.
+
+        """
+        new_data = {}
+        for block_name, block_dict in data.items():
+            new_data[block_name] = {}
+            # rebuild block_dict
+            for k, v in block_dict.items():
+                if k == 'data':
+                    new_data[block_name]['data'] = {}
+                    new_field_names = []
+                    for field_name, field_values in block_dict['data'].items():
+                        new_field_name = Provider._enforce_field_name_rules(field_name)
+                        new_field_name = Provider._check_for_duplicate_names(new_field_name,
+                                                                             new_field_names)
+
+                        new_data[block_name]['data'][new_field_name] = field_values
+
+                        new_field_names.append(new_field_name)
+                else:
+                    new_data[block_name][k] = v
+
+        return new_data
+
+    def save_to_block(self, data):
+        """Saves a list of data points into blocks. A block will be created
+        for any new block_name.
+
+        Examples:
+            The format of data is shown in the following example:
+
+            >>> data = {'test': {'block_name': 'test',
+                             'timestamps': [time.time()],
+                             'data': {'key1': [1],
+                                      'key2': [2]},
+                             'prefix': ''}
+                       }
+            >>> prov.save_to_block(data)
+
+            Note the block name shows up twice, once as the dict key in the
+            outer data dictionary, and again under the 'block_name' value -- in
+            this instance both the word 'test'.
```

Comment:
Not sure why this needs to be in the doc strings, but to answer your question the key is what's used in the function, and they should always match because the data structure is built in ocs_feed.py here: https://github.com/simonsobs/ocs/blob/74f9374bb0e5f61b51afb59a9b8a5119b25af690/ocs/ocs_feed.py#L127 
and is not set by the user.

block_name is used as both the dict key and is in Block.encoded(), so it appears twice.

Reply:
This wasn't clear to me when I was first trying to write a test for `save_to_block`. The messages that get passed to a feed are of the form given by `b.encoded()`, I think my assumption at the time was that `data` here followed the same form, but then I found it needed this additional block name, which is why I added it to this docstring.